base/algorithms.jl

```diff
@@ -0,0 +1,119 @@
+module Algorithms
+
+export replace!, replace
+
+import Base: replace
+
+
+"""
+    replace!(pred, [f::Function], A, [new], [count])
+
+Replace all occurrences `x` in collection `A` for which `pred(x)` is true
+by `new` or `f(x)` (exactly one among `f` and `new` must be specified).
+If `count` is specified, then replace at most `count` occurrences.
+This is the in-place version of [`replace`](@ref).
+
+# Examples
+```jldoctest
+julia> a = [1, 2, 3, 1];
+
+julia> replace!(isodd, a, 0, 2); a
+4-element Array{Int64,1}:
+ 0
+ 2
+ 0
+ 1
+
+julia> replace!(x->x.first=>3, Dict(1=>2, 3=>4), 1) do (k, v)
+           v < 3
+       end
+
+Dict{Int64,Int64} with 2 entries:
+  3 => 4
+  1 => 3
+```
+
+!!! note
+When `A` is an `Associative` or `AbstractSet` collection, if
+there are collisions among old and newly created keys, the result
+can be unexpected:
+
+```jldoctest
+julia> replace!(x->true, x->2x, Set([3, 6]))
+Set([12])
+```
+"""
+function replace!(pred, new::Function, A::AbstractArray, n::Integer=-1)
+    n == 0 && return A
```

Comment:
The `AbstractString` method uses `n=0` by default, so it means "unlimited". Your choice sounds better since it allows using `0` for a no-op, which can be useful if that's the result of a computation, but better adapt the existing method then. Would also make sense to check that `n > -1` to prevent unexpected things.